src/application/common/logger.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, unmanaged } from 'inversify';
+import { LoggerWatcher } from './logger-watcher';
+
+export namespace LogLevel {
+    export const FATAL = 60;
+    export const ERROR = 50;
+    export const WARN = 40;
+    export const INFO = 30;
+    export const DEBUG = 20;
+    export const TRACE = 10;
+}
+
+export const ILoggerServer = Symbol("ILoggerServer");
+
+export interface ILoggerServer {
+    setClient(client: ILoggerClient | undefined): void;
+    setLogLevel(id: number, logLevel: number): Promise<void>;
+    getLogLevel(id: number): Promise<number>;
+    log(id: number, logLevel: number, message: string, params: any[]): Promise<void>;
+    child(obj: Object): Promise<number>;
+}
+
+export const ILoggerClient = Symbol("ILoggerClient");
+
+export interface ILogLevelChangedEvent {
+    oldLogLevel: number;
+    newLogLevel: number;
+}
+
+export interface ILoggerClient {
+    onLogLevelChanged(event: ILogLevelChangedEvent): void;
+}
+
+export const ILogger = Symbol("ILogger");
+
+export interface ILogger {
+    child(obj: Object): ILogger;
+    setLogLevel(logLevel: number): Promise<void>
+    getLogLevel(): Promise<number>;
+    trace(message: string, ...params: any[]): void;
+    debug(message: string, ...params: any[]): void;
+    info(message: string, ...params: any[]): void;
+    warn(message: string, ...params: any[]): void;
+    error(message: string, ...params: any[]): void;
+    fatal(message: string, ...params: any[]): void;
+}
+
+@injectable()
+export class Logger implements ILogger {
+
+    /* Log level for the loggers.  */
+    private _logLevel: Promise<number>[] = [];
+
+    /* Root logger has id 1.  */
+    private rootLoggerId = 1;
```

Comment:
Indeed it is static. It does not belong to the instance but to the class.  In other words, if you create three logger instances, why would it have its own property `rootLoggerId`?